### ISSUE TITLE
Upgrade Javadoc plugin

### ISSUE THREAD

**Issue by laeubi** [eclipse-platform/eclipse.platform.releng.aggregator#1368]
We currently use an old version of javadoc plugin because of [MJAVADOC-707](https://issues.apache.org/jira/browse/MJAVADOC-707) but there is now a new release with a [flag called `legacyMode`](https://issues.apache.org/jira/browse/MJAVADOC-707?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17765906#comment-17765906) that should fix the issues we have seen.

Steps to perform:

- [ ] Upgrade the javadoc plugin and set the `legacyMode` flag
- [ ] deploy a local eclipse parent pom
- [ ] run the build of a subproject with javadoc enabled and check if everything works and javadocs are generated

**Comment by akurtakov** [comment]
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1369 would need legacyMode set as an extra change.

**Comment by laeubi** [comment]
Yes its just important to also check the results as in the past the javadoc build failed but the maven build suceeds so one could get the false impression everything is "fine"....

**Comment by akurtakov** [comment]
Local test with 3.6.0 and legacy mode made the *javadoc.jar properly generate. I'm pushing the changes.

**Comment by akurtakov** [comment]
Fixed via https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1369
